### PR TITLE
feat(ats): add ATS resume scoring with tournament-style NLP algorithms

### PR DESCRIPTION
**Pull request by athola**
Implement comprehensive ATS (Applicant Tracking System) resume scoring using multiple NLP algorithms for resume-job matching.

### Scoring Algorithms
- **TFIDFScorer**: Statistical term frequency and cosine similarity analysis
- **JaccardScorer**: N-gram phrase overlap detection (1-3 grams)
- **KeywordScorer**: Exact keyword matching with fuzzy tolerance (SequenceMatcher)
- **ATSTournament**: Combine multiple scorers with weighted averages (40/30/30 default weights)

### CLI Addition
- New \`screen\` command for terminal-based resume screening against job descriptions

### Supporting Components
- **EntityExtractor**: Structured data extraction (skills, experience, education, certifications)
- **ATSReportGenerator**: YAML/JSON/text report output with recommendations

### API Changes
- 13 new public API exports (see \`wiki/API-Reference.md\`)
- All exports follow stability contract (semver-guaranteed)

### Documentation
- \`wiki/ATS-Scoring-Rubric.md\`: Scoring methodology and formula reference
- \`wiki/API-Reference.md\`: Updated with ATS API documentation
- \`CHANGELOG.md\`: Version 0.2.0 changelog entry

## Testing
- 126 unit tests (all passing)
- Property-based tests with hypothesis
- Edge case coverage (empty inputs, unicode, special characters)
- API surface validation

## Version
- Bumped to 0.2.0 (minor version per semver for new feature)

---

*Review approved via \`/pr-review\` command. See review comments for details.*

### CODE REVIEW

**Comment by athola** [comment]
## Test Plan Verification

### Pre-Merge Testing Checklist

| Test Category | Tests | Status |
|--------------|-------|--------|
| Unit Tests (ATS Module) | 126 tests | ✅ PASS |
| Property-Based Tests | Included | ✅ PASS |
| API Surface Tests | 1 test | ✅ PASS |
| Linting (ruff) | All files | ✅ PASS |
| Version Consistency | pyproject.toml + __init__.py | ✅ PASS |

### Manual Verification Steps

Before merging, verify:

1. **CLI Command**
   ```bash
   simple-resume screen resume_private/input/my_resume.yaml "job_description.txt"
   ```
   Expected: YAML report with ATS scores

2. **Python API**
   ```python
   from simple_resume import score_resume, ATSTournament, TFIDFScorer
   
   result = score_resume(resume_text, job_description)
   assert 0 <= result.overall_score <= 1
   ```

3. **Report Generation**
   ```python
   from simple_resume import ATSReportGenerator
   
   gen = ATSReportGenerator(result, "resume.yaml", "job.yaml")
   yaml_report = gen.generate_yaml()
   json_report = gen.generate_json()
   ```

### Test Evidence

- 126 tests passed in 9.08s
- Includes property-based testing with hypothesis
- Edge cases covered: empty inputs, unicode, special characters
- API surface validated against `__all__` exports

### Coverage Note

Overall project coverage: 35% (ATS module specifically has good coverage, but total is below 85% target).

---

*Test plan posted via `/pr-review` command*

File: src/simple_resume/core/ats/entities.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from re import Pattern
+
+from simple_resume.core.ats.base import ExtractedEntities
+
+# Common technical skills patterns (extensible)
+TECH_SKILLS_PATTERNS = [
+    r"\b(?:Python|JavaScript|TypeScript|Java|C\+\+|C#|Go|Rust|Ruby|PHP|Swift|Kotlin)\b",
```
Comment (athola) [inline src/simple_resume/core/ats/entities.py]: is there instead some kind of online list we can pull these from as a primary and then use this as a backup?

File: src/simple_resume/core/ats/tfidf.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        # Get top keywords from resume (what they offer)
+        resume_keyword_indices = {
+            i for i, score in enumerate(resume_tfidf) if score > 0
+        }
+        top_resume_keywords = [
+            (feature_names[i], float(resume_tfidf[i]))
+            for i in sorted(
+                resume_keyword_indices, key=lambda x: resume_tfidf[x], reverse=True
+            )
+        ][:20]
```
Comment (athola) [inline src/simple_resume/core/ats/tfidf.py]: Same here; what is the significance of 20?

File: src/simple_resume/core/ats/tfidf.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        # Keyword density: proportion of job keywords found in resume
+        keyword_density = len(intersection) / len(job_indices) if job_indices else 0.0
+
+        # Experience relevance: weighted by TF-IDF scores of shared terms
+        # Heuristic: Divide by 10.0 to normalize to [0, 1]
+        # Rationale: Empirically, shared TF-IDF products rarely exceed 10.0
+        # even for long documents. This provides a conservative cap that prevents
+        # saturation while allowing meaningful differentiation.
+        shared_tfidf_sum = sum(resume_tfidf[i] * job_tfidf[i] for i in intersection)
+        experience_relevance = min(1.0, shared_tfidf_sum / 10.0)  # Normalize to [0, 1]
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/core/ats/tfidf.py]: Fixed - extracted to constant `EXPERIENCE_RELEVANCE_NORMALIZER = 10.0` with documentation explaining it normalizes the experience relevance score.

File: src/simple_resume/core/palettes/exceptions.py [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
-from __future__ import annotations
-
-
-class PaletteError(RuntimeError):
-    """Base class for palette-related failures."""
-
-
-class PaletteLookupError(PaletteError):
-    """Raised when a named palette cannot be located."""
+.. deprecated::
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/core/palettes/exceptions.py]: Fixed - removed the deprecated `palettes/exceptions.py` file and updated all imports across the codebase to use `simple_resume.core.exceptions` instead.

File: src/simple_resume/core/ats/entities.py [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+
+        Args:
+            text: Resume or job description text
+            **kwargs: Additional parameters
+
+        Returns:
+            ExtractedEntities with all extracted information
+
+        """
+        entities = ExtractedEntities()
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/core/ats/entities.py]: Created issue #60
